test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -2415,12 +2414,11 @@ private void updateMinMasterNodes(int eligibleMasterNodeCount) {
             // there should be at least one master to update
             logger.debug("updating min_master_nodes to [{}]", minMasterNodes);
             try {
-                assertAcked(
-                    client().admin()
-                        .cluster()
-                        .prepareUpdateSettings()
-                        .setTransientSettings(Settings.builder().put(DISCOVERY_ZEN_MINIMUM_MASTER_NODES_SETTING.getKey(), minMasterNodes))
-                );
+                client().admin()
+                    .cluster()
+                    .prepareUpdateSettings()
+                    .setTransientSettings(Settings.builder().put(DISCOVERY_ZEN_MINIMUM_MASTER_NODES_SETTING.getKey(), minMasterNodes))
+                    .get(TimeValue.timeValueSeconds(30));
```

Comment:
I am a bit confused.

This says zen2 is default, however the actual constant value is `zen`:

https://github.com/elastic/elasticsearch/blob/bcfbf000748b538884087623113682165abcb92d/server/src/main/java/org/elasticsearch/discovery/DiscoveryModule.java#L63-L72

Anyways, even if it is using the different discovery type, is it safe to ignore ack?
The property is still used in couple of places as for 7.17, such as https://github.com/elastic/elasticsearch/blob/bcfbf000748b538884087623113682165abcb92d/server/src/main/java/org/elasticsearch/gateway/Gateway.java#L44

Reply:
`zen` means Zen2, there is no option in 7.x to use the legacy discovery type.

Yes I believe this is safe, or at least it's no less safe than other things related to upgrading from 6.x. Which was never completely safe anyway.